internal/operators/lvm/lvm_operator.go

```diff
@@ -118,14 +124,33 @@ func (o *operator) ValidateHost(ctx context.Context, cluster *common.Cluster, ho
 		return api.ValidationResult{Status: api.Failure, ValidationId: o.GetHostValidationID(), Reasons: []string{message}}, err
 	}
 
-	// GetValidDiskCount counts the total number of valid disks in each host and return an error if we don't have the disk of required size
 	diskCount := o.getValidDiskCount(inventory.Disks, host.InstallationDiskID)
-	if err != nil {
-		return api.ValidationResult{Status: api.Failure, ValidationId: o.GetHostValidationID(), Reasons: []string{err.Error()}}, nil
+	mode := o.GetDeploymetMode(cluster)
+
+	role := common.GetEffectiveRole(host)
+
+	message := "Logical Volume Manager requires at least one non-installation HDD/SSD disk on the host"
+	if mode == common.ClusterDeploymentModeCompact || mode == common.ClusterDeploymentModeSNO {
+		if role == models.HostRoleMaster || role == models.HostRoleAutoAssign {
+			if diskCount == 0 {
+				return api.ValidationResult{Status: api.Failure, ValidationId: o.GetHostValidationID(), Reasons: []string{message}}, nil
+			}
+			return api.ValidationResult{Status: api.Success, ValidationId: o.GetHostValidationID()}, nil
+		}
+	}
+
+	if mode == common.ClusterDeploymentModeStandard {
+		if role == models.HostRoleAutoAssign {
+			status := "For LVM Standard Mode, host role must be assigned to master or worker."
```

Comment:
You can make this code much more straightforward:

```
validationStatus := api.Success
if common.GetEffectiveRole(host) == models.HostRoleWorker || common.AreMastersSchedulable(cluster) {
    if diskCount == 0 {
        validationStatus = api.Failure
    }
}
```

No need for all the consts of cluster types, the validations about autoassign, etc.